### PR TITLE
Feature/integ specs views

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
### Project requirements
- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
 
   - include the screenshots of your app's logs before and after the fix
   - before
        before

    - after
      after
-[x]  Integration specs
      -  Use Capybara to write integration tests for each view in your project.
test resu

### CODE REVIEW

File: spec/feature/posts_index_spec.rb [hunk 1]
```diff
@@ -47,14 +38,4 @@
   it 'Pagination' do
     expect(page).to have_button('Pagination')
   end
-
```
Comment (t-yanick) [inline spec/feature/posts_index_spec.rb:50]: - [ ] Great job working on this milestone. You are currently missing the test that `redirects to the post's show page when you click on a post.`

File: spec/feature/posts_index_spec.rb [hunk 2]
```diff
@@ -1,36 +1,27 @@
...
-    expect(page).to have_css("img[src='#{user.photo}'][alt='user photo']")
-    expect(page).to have_content(user.name)
-    expect(page).to have_content("Number of posts: #{user.post_counter}")
-  end
-
   it 'displays each post with relevant information' do
     @posts = user.posts
 
     @posts.each do |post|
-      expect(page).to have_content(post.title)
```
Comment (t-yanick) [inline spec/feature/posts_index_spec.rb:29]: - [ ] You are currently missing:
- Test that asserts you can see the number of posts the user has written.
- Test that asserts you can see a post's title.
- Test that asserts you can see some of the post's body.
Kindly add the missing tests

File: spec/feature/posts_index_spec.rb [hunk 3]
```diff
@@ -1,36 +1,27 @@
...
-        comments_counter: 3,
-        likes_counter: 7
+        comments_counter: 2,
+        likes_counter: 1
       )
     end
     visit user_posts_path(user)
   end
 
-  it 'displays user profile information' do
```
Comment (t-yanick) [inline spec/feature/posts_index_spec.rb:19]: - [ ] You are currently missing the following tests:
- Test that asserts you can see the user's profile picture.
- Test that asserts you can see the user's username.
Kindly add these missing test cases

File: spec/feature/posts_show_spec.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 require 'rails_helper'
 
 RSpec.describe 'Post show page', type: :feature do
-  let!(:user) { User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10) }
+  let!(:user) { User.create(name: 'Sarkodie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', posts_counter: 10) }
   let!(:post) do
     Post.create(
```
Comment (t-yanick) [inline spec/feature/posts_show_spec.rb:6]: - [ ]  At the moment you are missing the test that asserts that you can see who wrote the post. Kindly add it.

File: spec/feature/users_index_spec.rb [hunk 5]
```diff
@@ -25,12 +25,6 @@
     end
   end
 
-  it 'displays the number of posts each user has written' do
```
Comment (t-yanick) [inline spec/feature/users_index_spec.rb:28]: - [ ] You are currently missing the following test:
- Test that asserts you can see the number of posts each user has written.
Kindly add the missing test case

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Benawi , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Collaborative work 👍 Commits from both team members
:heavy_check_mark: Screenshots of N + 1 solution attached to PR description
:heavy_check_mark: Has Capybara gem in `Gemfile` for integration test
:heavy_check_mark: Most test cases added

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
### Changes 
- [x] Add test cases for redirects to the post's show page when you click on a post 
- [x] Issues(already written before)fixed related to 
       - Test that asserts you can see the number of posts the user has written.
       - Test that asserts you can see a post's title.
       - Test that asserts you can see some of the post's body
       - Test that asserts that you can see who wrote the post.
       - Test that asserts you can see the number of posts each user has written.

image
image

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Benawi 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
✔️Descriptive Pull Request
✔️ Screenshots of N + 1 solution attached to PR description
✔️ Has Capybara gem in Gemfile for integration test
✔️ test cases added
✔️No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**